app/services/ingest_iati_activities.rb

```diff
@@ -0,0 +1,175 @@
+    if accountable_org_element
+      accountable_org_reference = accountable_org_element.attributes["ref"].value
+      accountable_organisation = Organisation.find_by(iati_reference: accountable_org_reference)
+      new_activity.accountable_organisation_name = accountable_organisation.name
+      new_activity.accountable_organisation_type = accountable_organisation.organisation_type
+      new_activity.accountable_organisation_reference = accountable_organisation.iati_reference
+    end
+
+    # There is no DP reference in the extending element. We must ask the user for it.
+    new_activity.extending_organisation = delivery_partner
+
+    implementing_org_elements = legacy_activity.elements.select { |element| element.name.eql?("participating-org") && element.attributes["role"].value.eql?("4") }
+    implementing_org_elements.each do |org_element|
+      new_activity.implementing_organisations << ImplementingOrganisation.create!(
+        name: org_element.children.detect { |child| child.name.eql?("narrative") }.text,
+        organisation_type: org_element.attributes["type"].value,
+        activity: new_activity
+      )
+    end
+  end
+
+  private def add_transactions(legacy_activity:, new_activity:)
+    transaction_elements = legacy_activity.elements.select { |element| element.name.eql?("transaction") }
+    transaction_elements.each do |transaction_element|
+      currency = transaction_element.children.detect { |child| child.name.eql?("value") }.attributes["currency"].value
+      date = transaction_element.children.detect { |child| child.name.eql?("value") }.attributes["value-date"].value
+      value = transaction_element.children.detect { |child| child.name.eql?("value") }.children.text
+      transaction_type = transaction_element.children.detect { |child| child.name.eql?("transaction-type") }.attributes["code"].value
+
+      description = if transaction_element.children.detect { |child| child.name.eql?("description") }.present?
+        transaction_element.children.detect { |child| child.name.eql?("description") }.children.detect { |child| child.name.eql?("narrative") }.text
+      else
+        "Unknown description"
+      end
+
+      providing_organisation_name = transaction_element.children.detect { |child| child.name.eql?("provider-org") }.children.detect { |child| child.name.eql?("narrative") }.text
+      providing_organisation_reference = transaction_element.children.detect { |child| child.name.eql?("provider-org") }.attributes["ref"].value
+      receiving_organisation_name = transaction_element.children.detect { |child| child.name.eql?("receiver-org") }.children.detect { |child| child.name.eql?("narrative") }.text
+      receiving_organisation_reference = transaction_element.children.detect { |child| child.name.eql?("receiver-org") }.attributes["ref"].try(value, nil)
+
+      transaction = Transaction.new(
+        reference: SecureRandom.hex,
```

Comment:
Reference is being removed during other ongoing work so this sits here now as something we don't have to think about too hard.